gnucash/import-export/ofx/gnc-ofx-import.c

```diff
@@ -1086,6 +1086,12 @@ void gnc_file_ofx_import (GtkWindow *parent)
 
         DEBUG("Opening selected file");
         libofx_proc_file(libofx_context, selected_filename, AUTODETECT);
+        // Now would be a good time to see whether the view has anything in it!
+        if(gnc_gen_trans_list_empty(gnc_ofx_importer_gui))
+        {
+            gnc_gen_trans_list_delete (gnc_ofx_importer_gui);
+            gnc_info_dialog(parent,_("OFX file imported, no new transactions"));
+        }
```

Comment:
I really wonder if this is the right location to test for empty view.

1. The UI will flash the importer before showing the no-transaction info dialog
2. (I may be wrong here) The dialog is only shown on OFX import; other importers could use it. Do you think you can replicate functionality in `import-export`?

Reply:
Well you can't test until all the transactions have been processed, which occurs one line above:
```
libofx_proc_file(libofx_context, selected_filename, AUTODETECT);
```
I'm not seeing the UI flash. Does this command (a few lines above)
```
gnc_ofx_importer_gui = gnc_gen_trans_list_new (GTK_WIDGET(parent), NULL, FALSE, 42);
```
Causes the window to appear? Or simply creates it without showing it.
About 2) yes, I can try to replicate it in import-export. Let me take a look.

Reply:
mmm, I looked, and there isn't a whole lot of code reuse in import-export, I think I'd rather not touch anything.

Reply:
It does create it and show it, if you look at that function, near the bottom is a gtk_widget_show_all. I would think you could remove that line, create a new function in import_main_matcher to retrieve the main_widget and then add an else clause to your if statement to do the gtk_widget_show_all but you will need to find any other occurrence of gnc_gen_trans_list_new and make appropriate changes and this would be a separate commit on this PR.

Reply:
Ah thanks. Normally I would add an optional input parameter with a default to False, and use that, which would ensure back-compatibility but this is C code so that's not an option. Let me see how much I would have to change if I added that parameter.